src/Components/Carousel.tsx

```diff
@@ -291,14 +317,7 @@ export class BaseCarousel<T> extends React.Component<
           height={height}
           onClick={() => {
             this.flickity.previous()
-
-            if (onArrowClick) {
-              onArrowClick({
-                state: this.state,
-                props: this.props,
-                flickity: this.flickity,
-              })
-            }
+            setImmediate(this.handleArrowClick)
```

Comment:
Noticed this bug -- we need to wait for flickity's internal state to finish updating